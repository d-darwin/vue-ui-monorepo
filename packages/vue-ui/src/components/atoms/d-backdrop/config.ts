import { CssPosition } from "./types";
import styles from "./index.css?module";

export default {
  name: "DBackdrop",
  class: styles.dBackdrop,
  opacity: 0.5,
  zIndex: 1000,
  position: CssPosition.fixed,
} as const;
