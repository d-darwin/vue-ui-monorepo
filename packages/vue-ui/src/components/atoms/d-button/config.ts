export default {
  name: "DButton",
  className: "dButton",
};
