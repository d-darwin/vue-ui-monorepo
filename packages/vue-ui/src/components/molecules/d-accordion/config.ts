import dAccordionStyles from "./d-accordion.css";
import dDetailsStyles from "./d-details.css";

export default {
  name: "DAccordion",
  class: dAccordionStyles["dAccordion"],
  details: {
    ref: "detailsRef",
    name: "DDetails",
    class: dDetailsStyles["dDetails"],
  },
  summaryOptions: {
    class: dDetailsStyles["summary"],
  },
  summaryIcon: "â››",
  summaryAfterClass: dDetailsStyles["summaryAfter"],
  contentOptions: {
    key: "detailsContent",
    ref: "detailsContentRef",
    class: dDetailsStyles["content"],
  },
  provideInjectKey: "commonProps",
} as const;
