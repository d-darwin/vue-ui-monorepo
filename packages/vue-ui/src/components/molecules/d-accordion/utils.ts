import { Transition } from "@darwin-studio/ui-codegen/dist/types/transition";
import { TRANSITION_VALUE } from "@darwin-studio/ui-codegen/dist/constants/transition";
import getConstantKey from "@darwin-studio/vue-ui/src/utils/get-constant-key";

export function getTransitionDuration(transition?: Transition): number {
  if (!transition) return 0;

  const transitionKey = getConstantKey(
    transition
  ) as keyof typeof TRANSITION_VALUE;

  return TRANSITION_VALUE[transitionKey]?.duration || 0;
}
