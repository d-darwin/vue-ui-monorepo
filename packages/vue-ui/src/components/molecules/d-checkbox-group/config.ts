export default {
  name: "DCheckboxGroup",
  className: "dCheckboxGroup",
  labelClassName: "label",
  errorClassName: "error",
} as const;
