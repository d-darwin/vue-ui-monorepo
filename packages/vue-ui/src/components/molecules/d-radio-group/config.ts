import { TYPE } from "@darwin-studio/vue-ui/src/components/atoms/d-caption/constant";
import styles from "./index.css?module";

export default {
  name: "DRadioGroup",
  class: styles["dRadioGroup"],
  labelClass: styles["label"],
  radioClass: styles["radio"],
  captionOptions: {
    type: TYPE.DANGER,
    class: styles["caption"],
  },
  captionOffset: "0.2em",
} as const;
