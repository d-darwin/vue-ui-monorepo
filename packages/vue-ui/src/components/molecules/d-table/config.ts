export default {
  name: "DTable",
  className: "dTable",
  rowClassName: "row",
} as const;
