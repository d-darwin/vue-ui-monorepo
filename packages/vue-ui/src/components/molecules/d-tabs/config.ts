import { COLOR_SCHEME } from "@darwin-studio/ui-codegen/dist/constants/color-scheme";
import { SIZE } from "@darwin-studio/ui-codegen/dist/constants/size";
import { TAG_NAME } from "@darwin-studio/vue-ui/src/constants/tag-name";
import dTabsStyles from "./d-tabs.css?module";
import dTabStyles from "./d-tab.css?module";
import dTabpanelStyles from "./d-tabpanel.css?module";

export default {
  tabsName: "DTabs",
  tabsClass: dTabsStyles.dTabs,
  provideInjectKey: "tabsProvided",
  injectedActiveIdPath: "injection.activeId",
  colorScheme: COLOR_SCHEME.PRIMARY,
  size: SIZE.MEDIUM,

  tablistOptions: {
    role: "tablist",
    class: dTabsStyles.dTablist,
  },
  tablistTag: TAG_NAME.UL,
  tablistRef: "tablist",

  tabName: "DTab",
  tabOptions: {
    role: "tab",
    class: dTabStyles.dTab,
  },
  tabTag: TAG_NAME.LI,

  tabpanelName: "DTabpanel",
  tabpanelOptions: {
    role: "tabpanel",
    tabindex: 0,
    class: dTabpanelStyles.dTabpanel,
  },
  tabpanelTag: TAG_NAME.DIV,
};
