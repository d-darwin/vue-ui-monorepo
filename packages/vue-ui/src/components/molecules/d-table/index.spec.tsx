import { shallowMount } from "@vue/test-utils";
import { baseClassCase } from "@/utils/test-case-factories";
import DTable from "./index";
import config from "./config";

const head0Label = "head 1";
const head1Label = "head 2";
const head2Label = "head 3";
const headRows = [
  [
    head0Label,
    <b>{head1Label}</b>,
    <div class="inner-th-element">{head2Label}</div>,
  ],
];

describe("DTable", () => {
  const wrapper = shallowMount(DTable);

  baseClassCase(wrapper, config.className);

  it("Should render props.headClass to the thead element class", async () => {
    const headClass = "some-head-class";
    await wrapper.setProps({ headClass });
    const theadEl = wrapper.find("thead");
    expect(theadEl).toBeTruthy();
    expect(theadEl.classes()).toContain(headClass);
  });

  it("Should render props.headRows as tr elements inside the thead", async () => {
    await wrapper.setProps({ headRows });
    const thElements = wrapper.findAll("thead > tr > th");
    expect(thElements.length).toBe(headRows[0].length);
    expect(thElements[0].text()).toBe(head0Label);
    expect(thElements[1].text()).toBe(head1Label);
    expect(thElements[2].html()).toMatchSnapshot();
  });

  it("Should render props.headRowClass as tr elements inside the thead class", async () => {
    const headRowClass = "some-head-row-class";
    await wrapper.setProps({ headRows, headRowClass });
    const headRowEl = wrapper.find("thead > tr");
    expect(headRowEl.classes()).toContain(headRowClass);
  });

  it("Should render props.headCellClass as td elements inside the thead class", async () => {
    const headCellClass = "some-head-row-class";
    await wrapper.setProps({ headRows, headCellClass });
    const thElements = wrapper.findAll("thead > tr > th");
    thElements.forEach((thEl) => {
      expect(thEl.classes()).toContain(headCellClass);
    });
  });

  it("Should render props.headRowAttrs as tr elements inside the thead attrs", async () => {
    const headRowAttrsObj = {
      tabindex: "1",
      "data-id": "123",
    };
    await wrapper.setProps({ headRows, headRowAttrs: () => headRowAttrsObj });
    const headRowEl = wrapper.find("thead > tr");
    expect(headRowEl.attributes("tabindex")).toBe(headRowAttrsObj.tabindex);
    expect(headRowEl.attributes("data-id")).toBe(headRowAttrsObj["data-id"]);
  });

  it("Should render props.headCellAttrs as td elements inside the thead attrs", async () => {
    const headCellAttrsObj = {
      style: "background: red;",
      onClick: jest.fn(),
    };
    await wrapper.setProps({ headRows, headCellAttrs: () => headCellAttrsObj });
    const thElements = wrapper.findAll("thead > tr > th");
    await Promise.all(
      thElements.map(async (thEl) => {
        expect(thEl.attributes("style")).toBe(headCellAttrsObj.style);
        await thEl.trigger("click");
      })
    );
    expect(headCellAttrsObj.onClick).toBeCalledTimes(headRows[0].length);
  });

  /*
  it("Should render props.bodyClass to the tbody element class", async () => {
    const bodyClass = "some-body-class";
    await wrapper.setProps({ bodyClass });
    const theadEl = wrapper.find("tbody");
    expect(theadEl).toBeTruthy();
    expect(theadEl.classes()).toContain(bodyClass);
  });

  // TODO: props.bodyRows
  it("Should render props.bodyRows as tr elements inside the tbody", async () => {
    expect(true).toBeFalsy();
  });
  // TODO: props.bodyRowClass
  it("Should render props.bodyRowClass as tr elements inside the tbody class", async () => {
    expect(true).toBeFalsy();
  });
  // TODO: props.bodyCellClass
  it("Should render props.bodyCellClass as td elements inside the tbody class", async () => {
    expect(true).toBeFalsy();
  });
  // TODO: props.bodyRowAttrs
  it("Should render props.bodyRowAttrs as tr elements inside the tbody attrs", async () => {
    expect(true).toBeFalsy();
  });
  // TODO: props.bodyCellAttrs
  it("Should render props.bodyCellAttrs as td elements inside the tbody attrs", async () => {
    expect(true).toBeFalsy();
  });

  // TODO: props.padding
  it("Should render props.padding as td elements padding class", async () => {
    expect(true).toBeFalsy();
  });
  // TODO: props.size
  it("Should render props.bodyCellAttrs as td elements size class", async () => {
    expect(true).toBeFalsy();
  });
  // TODO: props.transition
  it("Should render props.bodyCellAttrs as td elements transition transition", async () => {
    expect(true).toBeFalsy();
  });

  // TODO: props.enableHtml
  it("props.enableHtml should allow to pass HTML strings as content of the td elements", async () => {
    expect(true).toBeFalsy();
  });*/
});
