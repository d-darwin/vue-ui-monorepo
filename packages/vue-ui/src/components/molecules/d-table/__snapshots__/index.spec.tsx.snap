// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DTable Should render props.headRows as tr elements inside the thead 1`] = `
<th class="paddingDefaultMedium paddingDefault sizeMedium">
  <div class="inner-th-element">head 3</div>
</th>
`;
